src/merkle/simple_smt/mod.rs

```diff
@@ -154,27 +149,25 @@ impl SimpleSmt {
     pub fn insert_leaf(&mut self, key: u64, value: Word) -> Result<(), MerkleError> {
         self.store.insert_leaf_node(key, value);
 
-        let depth = self.depth();
-        let mut curr_key = key;
-        let mut curr_node: RpoDigest = value.into();
-        for n in (0..depth).rev() {
-            let parent_key = curr_key >> 1;
-            let parent_node = self
+        // TODO consider using a map `index |-> word` instead of `index |-> (word, word)`
+        let mut node = NodeIndex::new(self.depth(), key);
```

Comment:
nit: should the variable be named `index` rather than `node`?

Reply:
Done!